arangod/Agency/Node.cpp

```diff
@@ -495,11 +505,17 @@ ResultT<std::shared_ptr<Node>> Node::handle<INCREMENT>(VPackSlice const& slice)
   size_t inc = (slice.hasKey("step") && slice.get("step").isUInt())
                    ? slice.get("step").getUInt() : 1;
   Builder tmp;
+  bool applied = false;
   {
     VPackObjectBuilder t(&tmp);
     try {
-      tmp.add("tmp", Value(this->slice().getInt() + inc));
-    } catch (std::exception const&) {
+      if (!lifetimeExpired()) {
+        tmp.add("tmp", Value(this->slice().getInt() + inc));
+        applied = true;
+      }
+    } catch (...) {}
+
+    if (!applied) {
       tmp.add("tmp", Value(1));
     }
```

Comment:
If there's an exception thrown in the `try{}` part, it's not obvious to me that the builder is in a well-defined state and we can just continue adding things. I'd rather let the exception throw, unless you know something here I don't.

Reply:
Same holds for the other case below.

Reply:
@kvahed could you comment on this one?